### PR TITLE
fixed correctness for soft delete and MUA

### PR DESCRIPTION
**Pull request by hiaga**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [public repo review queue], [private repo review queue]  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue]  

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [x] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[public repo review queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo review queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AWaitForARMFeedback+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AIDCDevDiv+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by hiaga** [comment]
adding the correctness fix for older API version 2023-02-01 like we did for API version 2023-04-01  - https://github.com/kalyanachakravarthikarri-az/azure-rest-api-specs-forked/blob/6ccb8d1cf74dfa4885ad371af238c29617fcb2d0/specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-04-01/vaults.json

**Comment by TimLovellSmith** [comment]
Please complete the mandatory PR questionnaire including purpose of PR and due diligence.
Please confirm whether this non-preview, so GA, API has already shipped to customers.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json [hunk 1]
```diff
@@ -1190,6 +1190,44 @@
...
+    "SoftDeleteSettings": {
+      "description": "Soft delete Settings of vault",
+      "type": "object",
+      "properties": {
+        "softDeleteState": {
+          "enum": [
+            "Invalid",
+            "Enabled",
+            "Disabled",
+            "AlwaysON"
```
Comment (TimLovellSmith) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1202]: nit, should it be capitalized 'AlwaysOn'? AlwaysON isn't the usual for PascalCase.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json [hunk 2]
```diff
@@ -1190,6 +1190,44 @@
...
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SoftDeleteState",
+            "modelAsString": true
+          }
+        },
+        "softDeleteRetentionPeriodInDays": {
+          "format": "int32",
+          "description": "Soft delete retention period in days",
+          "type": "integer"
```
Comment (TimLovellSmith) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1213]: consider specifying min or max values if they apply

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json [hunk 3]
```diff
@@ -1190,6 +1190,44 @@
...
+    },
+    "MultiUserAuthorization": {
+      "description": "MUA Settings of vault",
+      "enum": [
+        "Invalid",
+        "Enabled",
+        "Disabled"
+      ],
+      "type": "string",
+      "readOnly": true,
```
Comment (TimLovellSmith) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1225]: I don't think readOnly means anything in the context of an enum definition?

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json [hunk 4]
```diff
@@ -1190,6 +1190,44 @@
...
+      "x-ms-enum": {
+        "name": "MultiUserAuthorization",
+        "modelAsString": true
```
Comment (TimLovellSmith) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1228]: please fully document the meaning of all these enum values and for instance, what is the semantics of Invalid as compared to Disabled?

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json (hunk 1 above)
Comment (TimLovellSmith) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1202]: (But by all means make it match your backends current behavior)

**Comment by hiaga** [comment]
> Please complete the mandatory PR questionnaire including purpose of PR and due diligence. Please confirm whether this non-preview, so GA, API has already shipped to customers.

@TimLovellSmith : this is a S360 swagger quality fix for an older API version. Later stable versions already have this fix - 
https://github.com/Azure/azure-rest-api-specs/pull/24211/files#diff-23c7e09767e1c7b29e567b0fbe0c06ca6d04f326df34f1e1d27130bb96a2ae61

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json (hunk 1 above)
Comment (hiaga) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1202]: this is as per our service

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json (hunk 3 above)
Comment (hiaga) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2023-02-01/vaults.json:1225]: The value for MultiUserAuthorization can't be set and is readonly. Also this is how it was added in later APi verisons - https://github.com/Azure/azure-rest-api-specs/pull/24211/files#diff-23c7e09767e1c7b29e567b0fbe0c06ca6d04f326df34f1e1d27130bb96a2ae61 

should I still remove it ?

**Pull request closed without merging**